appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/PoAManager.java

```diff
@@ -208,34 +210,30 @@ private void sendProof() {
   }
 
   private void handleCampaign() {
-    if (hasNetwork()) {
-      String packageName = appContext.getPackageName();
-      Disposable subscribe = Single.fromCallable(() -> getVerCode(appContext, packageName))
-          .subscribeOn(Schedulers.io())
-          .map(verCode -> getActiveCampaigns(packageName, BigInteger.valueOf(verCode)))
-          .subscribe(campaigns -> {
-            if (campaigns.isEmpty()) {
-              stopProcess();
-            } else {
-              BigInteger campaignId = campaigns.get(0)
-                  .getId();
-
-              Bundle bundle = new Bundle();
-              bundle.putString("packageName", appContext.getPackageName());
-              bundle.putString("campaignId", campaignId.toString());
-
-              poaConnector.sendMessage(appContext, MSG_REGISTER_CAMPAIGN, bundle);
-
-              this.campaignId = campaignId;
-            }
-          });
-    }
-  }
-
-  private boolean hasNetwork() {
-    NetworkInfo activeNetworkInfo = ((ConnectivityManager) appContext.getSystemService(
-        Context.CONNECTIVITY_SERVICE)).getActiveNetworkInfo();
-    return activeNetworkInfo != null && activeNetworkInfo.isConnected();
+    String packageName = appContext.getPackageName();
+
+    compositeDisposable.add(ReactiveNetwork.observeInternetConnectivity()
+        .subscribeOn(Schedulers.io())
+        .filter(aBoolean -> aBoolean)
```

Comment:
What is the meaning of this "aBoolean"? Is it possible to be more self explanatory?